pkg/lib/operation/project/remote/table/import/operation.go

```diff
@@ -115,3 +124,59 @@ func checkTableExists(ctx context.Context, d dependencies, tableKey keboola.Tabl
 	}
 	return true
 }
+
+func getColumns(o Options) (keboola.Columns, error) {
+	var c keboola.Columns
+	var err error
+	if o.Columns != nil && o.WithoutHeaders {
+		c = convertHeadersToColumns(o.Columns)
+	}
+
+	if o.Columns == nil && !o.WithoutHeaders {
+		c, err = extractColumnsFromCsv(o.FileName)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return c, nil
+}
+
+// this function converts array string to keboola.Columns.
+func convertHeadersToColumns(headers []string) keboola.Columns {
+	var columns keboola.Columns
+	for _, header := range headers {
+		columns = append(columns, keboola.Column{Name: header})
+	}
+	return columns
+}
+
+// extractColumnsFromCsv returns a first row in the csv file (convertHeadersToColumns)
+// this function will be used if flags --convertHeadersToColumns and --file-without-headers aren't used and table doesn't exist.
+func extractColumnsFromCsv(f string) (keboola.Columns, error) {
+	// nolint: forbidigo
+	file, err := os.Open(f)
+	if err != nil {
+		return nil, err
+	}
+
+	defer file.Close()
+
+	// Create a new CSV reader
+	reader := csv.NewReader(file)
+
+	// Read the first line (headers)
+	headers, err := reader.Read()
+	if err != nil {
+		return nil, err
+	}
+
+	return convertHeadersToColumns(headers), nil
+}
+
+func printInfo(ctx context.Context, o Options, d dependencies, isCreated bool) {
```

Comment:
This is antipattern function. Based on `isCreated` it prints something, or something else, just extract it into code. Also it has single usage.